h/notification/mention.py

```diff
@@ -0,0 +1,16 @@
+import logging
+from dataclasses import dataclass
+
+from h.models import Annotation, Document, User
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Notification:
```

Comment:
This is h.notification.mention.Notification which is very similar to h.notification.reply.Notification.

Typing (mypy) will know the difference between the two but we tend to name things a bit more explicitly for "grepability" eg `MentionNotification`. 

Then this will become  `h.notification.mention.MentionNotification` which is more verbose but a bit clearer IMO.